### PR TITLE
feature -> feature | Updated: Forgot password page or reset password page

### PR DESCRIPTION
**Pull request by wri-haresh-p001**
I have created two vue pages name is forgot password page and reset password page or enter the email address send to verification code to registered mail Id and reset password page enter the verification code or enter the new password and save the database of new password.

Video URL :- https://img-v4.getdemo.dev/screenshot/chrome_ftJPmCoOQQ.mp4
Task URL :- https://team.webreinvent.com/backend/collab/project/vaahcms/tasks/186

### CODE REVIEW

File: Vue/vaahtwo/pages/public/ResetPassword.vue [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                        :loading="root.is_reset_password_btn_loading"/>
+                    <router-link :to="{name:'sign.in'}">
+                        <Button label="Sign In" class="p-button-text p-button-sm"/>
+                    </router-link>
+                </div>
+            </div>
+        </template>
+        <template #footer>
+            <div class="text-xs text-center"><p>
+                © 2022.
```
Comment (themodernpk) [inline Vue/vaahtwo/pages/public/ResetPassword.vue]: @we-vikram-wri231 such information should be dynamic and you can create a component which can be included wherever needed

File: Vue/vaahtwo/pages/public/ResetPassword.vue [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                        class="w-full"
+                        inputClass="w-full"
+                        toggleMask id="confirm-password"></Password>
+                    <label for="confirm-password">Confirm Password</label>
+                </div>
+                <div class="field flex justify-content-between align-items-center">
+                    <Button
+                        label="Recover"
+                        class="p-button-sm"
+                        @click="root.onResetPassword()"
```
Comment (themodernpk) [inline Vue/vaahtwo/pages/public/ResetPassword.vue]: @we-vikram-wri231 `onResetPassword` should not be the part of `root` store instead, you should create `auth` store. Anything related to `signin` `signup` and `forget` password should be the part of `auth` store

File: Vue/vaahtwo/pages/public/ResetPassword.vue [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                    <router-link :to="{name:'sign.in'}">
+                        <Button label="Sign In" class="p-button-text p-button-sm"/>
+                    </router-link>
+                </div>
+            </div>
+        </template>
+        <template #footer>
+            <div class="text-xs text-center"><p>
+                © 2022.
+                <a class="text-blue-400" href="https://vaah.dev/cms" target="_blank">VaahCMS</a>
```
Comment (themodernpk) [inline Vue/vaahtwo/pages/public/ResetPassword.vue]: @we-vikram-wri231 these links should also be dynamic, they are available in root assets and coming from `config/vaahcms.php` file

Comment (themodernpk) [reply Vue/vaahtwo/pages/public/ResetPassword.vue]: same feedback for `docs` link

File: Vue/vaahtwo/stores/root.js [hunk 4]
```diff
@@ -43,6 +53,54 @@ export const useRootStore = defineStore({
 
             }
         },
+        //-----------------------------------------------------------------------
+        onSendCode()
```
Comment (themodernpk) [inline Vue/vaahtwo/stores/root.js]: @we-vikram-wri231 we don't have a culture to prefix `on` , look for reference code

**Review by themodernpk — CHANGES_REQUESTED** [review]

File: Vue/vaahtwo/stores/root.js (hunk 4 above)
Comment (we-vikram-wri231) [reply Vue/vaahtwo/stores/root.js]: Yes Sir, I will make sure that these changes should be done.

**Pull request merged**